src/lib/validator.js

```diff
@@ -33,14 +33,18 @@ const fulfilTransfer = (request) => {
   const maxCallbackTimeLagDilation = Config.MAX_CALLBACK_TIME_LAG_DILATION_MILLISECONDS ? Config.MAX_CALLBACK_TIME_LAG_DILATION_MILLISECONDS : 0
   const completedTimestamp = new Date(request.payload.completedTimestamp)
   const now = new Date()
-  Logger.isDebugEnabled && Logger.debug(`completedTimestamp: ${completedTimestamp.toISOString()}, now: ${now.toISOString()}, maxLag: ${maxLag}, maxCallbackTimeLagDilation: ${maxCallbackTimeLagDilation}`)
+  Logger.isDebugEnabled && isValidDateObject(completedTimestamp) && Logger.debug(`completedTimestamp: ${completedTimestamp.toISOString()}, now: ${now.toISOString()}, maxLag: ${maxLag}, maxCallbackTimeLagDilation: ${maxCallbackTimeLagDilation}`)
```

Comment:
Why do we validate _completedTimestamp_ only before logging?  I mean, we use _completedTimestamp.getTime()_ on lines 37 and 39 as well.

Maybe, it's better to validate `request.payload.completedTimestamp`?  I see we have Joi request validation:
https://github.com/mojaloop/ml-api-adapter/blob/36f3755cb9f1a0b10a2efc1056f0a3ddea65bb36/src/api/transfers/routes.js#L127
 
 but Joi has appropriate validation for ISO 8601 date-time strings:
https://joi.dev/api/?v=17.9.1#stringisodate

Reply:
Thanks @geka-evk. Code updated. The extra validation was added mainly to cater for JS nuances with the date object and not necessarily the user input validation which is already taken care of by Joi validation. Since `completedTimestamp` is not required and JS allows initialization of Date with invalid or `undefined` or `null` input, that validation was necessary, but the same is achieved by just checking `request.payload.completedTimestamp` for nullity.